File: sdkwork-ai-sdk-go/api/audio.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type AudioApi struct {
    client *sdkhttp.Client
}

func NewAudioApi(client *sdkhttp.Client) *AudioApi {
    return &AudioApi{client: client}
}

// Create translation
func (a *AudioApi) CreateTranslation(body *sdktypes.CreateTranslationRequest, query map[string]interface{}) (sdktypes.TranscriptionResponse, error) {
    raw, err := a.client.Post(AiApiPath("/audio/translations"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.TranscriptionResponse
        return zero, err
    }
    return decodeResult[sdktypes.TranscriptionResponse](raw)
}

// Create transcription
func (a *AudioApi) CreateTranscription(body *sdktypes.CreateTranscriptionRequest, query map[string]interface{}) (sdktypes.TranscriptionResponse, error) {
    raw, err := a.client.Post(AiApiPath("/audio/transcriptions"), body, query, nil, "multipart/form-data")
    if err != nil {
        var zero sdktypes.TranscriptionResponse
        return zero, err
    }
    return decodeResult[sdktypes.TranscriptionResponse](raw)
}

// Create speech
func (a *AudioApi) CreateSpeech(body sdktypes.SpeechRequest) (string, error) {
    raw, err := a.client.Post(AiApiPath("/audio/speech"), body, nil, nil, "")
    if err != nil {
        var zero string
        return zero, err
    }
    return decodeResult[string](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/audio.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { SpeechRequest, TranscriptionResponse } from '../types';


export class AudioApi2 {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create translation */
  async createTranslation(body?: FormData, params?: QueryParams): Promise<TranscriptionResponse> {
    return this.client.post<TranscriptionResponse>(aiApiPath(`/v1/audio/translations`), body, params);
  }

/** Create transcription */
  async createTranscription(body?: FormData, params?: QueryParams): Promise<TranscriptionResponse> {
    return this.client.post<TranscriptionResponse>(aiApiPath(`/v1/audio/transcriptions`), body, params);
  }

/** Create speech */
  async createSpeech(body: SpeechRequest): Promise<string> {
    return this.client.post<string>(aiApiPath(`/v1/audio/speech`), body);
  }
}

export function createAudioApi2(client: HttpClient): AudioApi2 {
  return new AudioApi2(client);
}

export { AudioApi2 as AudioApi, createAudioApi2 as createAudioApi };
